src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerListener.java

```diff
@@ -94,31 +93,29 @@ private void updateDownstreamBuildAction(Run<?, ?> run) {
                 BuildUpstreamCause buildUpstreamCause = (BuildUpstreamCause) cause;
                 Run<?, ?> upstream = buildUpstreamCause.getUpstreamRun();
                 if (upstream instanceof FlowExecutionOwner.Executable) {
-                    FlowExecutionOwner owner = ((FlowExecutionOwner.Executable) upstream).asFlowExecutionOwner();
-                    if (owner == null) {
-                        LOGGER.log(Level.FINE, () -> "Unable to update DownstreamBuildAction for " + upstream + " node " + buildUpstreamCause.getNodeId());
-                        continue;
+                    DownstreamBuildAction action = findMatchingAction(upstream, buildUpstreamCause.getNodeId());
+                    if (action == null) {
+                        action = new DownstreamBuildAction(buildUpstreamCause.getNodeId(), run.getParent());
+                        run.addAction(action);
```

Comment:
Did you mean to add this to `upstream`? IIUC `run` is the downstream build (perhaps best to rename it for clarity)?